lib/rules/array-callback-return.js

```diff
@@ -188,19 +193,36 @@ module.exports = {
          * @returns {void}
          */
         function checkLastSegment(node) {
-            if (funcInfo.shouldCheck &&
-                funcInfo.codePath.currentSegments.some(isReachable)
-            ) {
-                context.report({
-                    node,
-                    loc: getLocation(node, context.getSourceCode()).loc.start,
-                    messageId: funcInfo.hasReturn
-                        ? "expectedAtEnd"
-                        : "expectedInside",
-                    data: {
-                        name: astUtils.getFunctionNameWithKind(funcInfo.node)
+            const method = astUtils.getStaticPropertyName(funcInfo.node.parent.callee);
```

Comment:
This will work most of the time, but this rule also checks cases like this:

```js
foo.map(bar || function () { /* this is also a callback */ });
```

It seems that we should store the method's name in `funcInfo`.